pkg/controller/seed-controller-manager/constraint-template-controller/controller.go

```diff
@@ -186,36 +181,40 @@ func (r *reconciler) syncAllClusters(
 	}
 
 	for _, userCluster := range clusterList.Items {
-		if userCluster.Status.NamespaceName == "" {
-			log.Debugw("Cluster has no namespace name yet, skipping", "cluster", userCluster.Spec.HumanReadableName)
+		clusterLog := log.With("cluster", userCluster.Name)
+
+		if userCluster.Spec.Pause {
+			clusterLog.Debug("Cluster is paused, skipping")
 			continue
 		}
 
-		if userCluster.Spec.Pause {
-			log.Debugw("Cluster paused, skipping", "cluster", userCluster.Spec.HumanReadableName)
+		// if the control plane is not healthy, we cannot possibly create a functioning usercluster client
+		if userCluster.Status.ExtendedHealth.ControlPlaneHealthy() {
```

Comment:
will the Cluster getting healthy trigger the watch event? Now when we have proper Cluster status I am not sure? In that case the CT wont get reconciled unless there is some cluster spec or CT change

Reply:
Excellent question. I just checked by [stalking](https://github.com/xrstf/stalk) a Cluster object and editing its status to set a new reconcile time. This then triggered a watch event with this diff:

![screenshot-2022-02-25-145816](https://user-images.githubusercontent.com/127499/155727683-18ab2e26-ec3c-42fa-9686-623edb914a27.png)

Reply:
awesome! Btw love the [stalk](https://github.com/xrstf/stalk)